client/src/main/java/tds/session/Extern.java

```diff
@@ -0,0 +1,45 @@
+package tds.session;
+
+/**
+ * configuration properties for the session system
+ */
+public class Extern {
+    private String clientName;
+    private String environment;
+
+    public static class Builder {
```

Comment:
We decided on using our best judgement when using a builder vs constructor.  At this time we'll leave this immutable and use a two argument constructor rather than builder.  Also, for Spring/Jackson to work properly I had to add a package private constructor.